### ISSUE TITLE
Docker fails to execute docker-entrypoint.sh if an image was built in Windows

### ISSUE THREAD

**Issue by dobromyslov** [n8n-io/n8n#5620]
**Describe the bug**

1. Some developers on Windows use the [auto CRLF option in Git](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings?platform=windows), and Git converts line endings to CRLF.
2. `docker-entrypoint.sh` files are also affected by such conversion.
3. Docker image built on Windows can't run `docker-entrypoint.sh` with CRLF line endings:
```
[FATAL tini (7)] exec /docker-entrypoint.sh failed: No such file or directory
```

**To Reproduce**

1. Use Windows
2. Enable the auto CRLF option with `git config --global core.autocrlf true`
5. Git clone the n8n repository (Git converts all line endings to CRLF)
6. Build the docker image
7. Run the docker image
8. See the error `[FATAL tini (7)] exec /docker-entrypoint.sh failed: No such file or directory`.

**Expected behavior**

I expect 
* the n8n repo to have its own settings for line endings 
* and Git not to convert line endings of the `.sh` files.

We can achieve this with the following config in the `.gitattributes` file:
```
*.sh text eol=lf
```

**Environment (please complete the following information):**

- OS: Windows
- n8n docker image built on Windows

**Workaround**
```
git config --global core.autocrlf input
git rm --cached -r .
git reset --hard
```

**Comment by Joffcom** [comment]
Hey @dobromyslov,

Thanks for reporting this one, It looks like a fairly quick fix. I will have a chat with the team to see if there is any possible downside to doing this.

**Comment by sdghf** [comment]
Hey @Joffcom I have also encountered this problem. After packaging the image on Windows and Mac, I uploaded it to my own image repository. However, I also encountered errors when running it through Potainer. I also tried to build and run the image using Drone, but encountered errors. Additionally, our own image size seems to be much smaller than the images on Docker Hub, so I'm trying to understand how you package the images.

**Comment by sdghf** [comment]
Hey @Joffcom  
This is a problem that occurs when running on Portainer, but there is no error when using the official image.
![image](https://user-images.githubusercontent.com/33833181/225504565-48cc8d24-ed2d-4779-bd8d-2901272dde3c.png)
![image](https://user-images.githubusercontent.com/33833181/225506234-5bb3bf96-8689-47f6-aa06-33090e5883fe.png)

**Comment by Joffcom** [comment]
Hey @sdghf 

Did you try the fix mentioned? Rather than using portainer what happens if you just use the docker run command?

Have you also checked that your build is creating an image for the correct cpu type? I know if I make a custom image on my M1 Mac it won't work on my nuc because I am not building an amd64 image.

I would also expect a custom image to be a lot bigger like 700mb - 1GB, can you share the exact command you are using to build the image and any changes you have made that require a custom image.

**Comment by sdghf** [comment]
Hey @Joffcom 

If running docker run locally is not an issue, but I want to deploy it to a server. I have tried to fix it, but the error still occurs. 

I know it's probably a CPU type issue, but I don't know how to build the correct image for the CPU type.

I packaged the image using docker/images/n8n-custom. I made some simple code modifications to adapt it to our own business, so I must create a custom image.

**Comment by Joffcom** [comment]
Hey @sdghf,

Can you confirm what CPU you are building on and what CPU you are deploying on? If you are making a custom image for another CPU you would need to use something buildx to pass in the platforms but that is not something I have instructions for and you would need to do a bit of digging through the docker docs for that one.

**Comment by Joffcom** [comment]
Hey @dobromyslov,

I have just popped in a PR which should fix this once merged, Thanks for raising this one. For now I am going to mark this one as closed if you still find it is an issue let me know and we can open it again.

**Comment by janober** [comment]
Fix got released with [`n8n@0.234.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.234.0)
